src/lib/degradeableItems.ts

```diff
@@ -302,51 +302,68 @@ export async function degradeItem({
 	const newCharges = Math.floor(currentCharges - chargesToDegrade);
 
 	if (newCharges <= 0) {
-		// If no more charges left, break and refund the item.
+		// If the user runs out for charges for a degradeable item, break and refund the item.
 		const hasEquipped = user.gear[degItem.setup].hasEquipped(item.id, false);
-		const hasInBank = user.owns(item.id);
+		const hasInBank = user.hasEquippedOrInBank(item.id);
+		const itemsDeleted = new Bank();
+
+		// Update the settingsKey to 0 to prevent negative charges
 		await user.update({
 			[degItem.settingsKey]: 0
 		});
-		const itemsDeleted = new Bank().add(item.id);
-
-		updateBankSetting('degraded_items_cost', itemsDeleted);
 
 		if (hasEquipped) {
-			// Get the users equipped item.
+			// Get the users equipped item and check for variants
 			let refundItems = degItem.itemsToRefundOnBreak;
-			for (const variant of degItem.refundVariants) {
-				if (user.gear[degItem.setup].hasEquipped(variant.variant.id, false, false)) {
-					refundItems = variant.refund;
+			let removeItems = degItem.item.id;
+			for (const variant of degItem.itemVariants!) {
+				if (user.gear[degItem.setup].hasEquipped(variant.chargedVariant.id, false, false)) {
+					removeItems = variant.chargedVariant.id;
+					refundItems = new Bank().add(variant.unchargedVariant).freeze();
+					break;
 				}
 			}
-			// Unequip and delete the users item.
+			// Unequip and delete the item from the users gear setup
 			const gear = { ...user.gear[degItem.setup].raw() };
 			gear[item.equipment!.slot] = null;
 			await user.update({
 				[`gear_${degItem.setup}`]: gear
 			});
-			// Give the user the uncharged version of their charged item.
+			console.log(removeItems);
```

Comment:
left some logging in unsure if that was intentional?

Reply:
fixed